config/v1/types_network.go

```diff
@@ -13,20 +13,50 @@ type Network struct {
 	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// spec holds user settable values for configuration
+	// Spec holds user settable values for configuration.
 	Spec NetworkSpec `json:"spec"`
-	// status holds observed values from the cluster. They may not be overridden.
+	// Status holds observed values from the cluster. They may not be overridden.
 	Status NetworkStatus `json:"status"`
 }
 
+// NetworkSpec is the desired network configuration.
+// As a general rule, this SHOULD NOT be used directly. Instead, you should
+// consume the NetworkStatus, as it indicates the currently deployed configuration.
 type NetworkSpec struct {
-	// serviceCIDR
-	// servicePortRange
-	// vxlanPort
-	// 	ClusterNetworks    []ClusterNetworkEntry `json:"clusterNetworks"`
+	// IP address pool to use for pod IPs.
+	ClusterNetworks []ClusterNetwork `json:"clusterNetworks"`
```

Comment:
So, totally nitpicking since this may be the last chance to change these...

Technically there is only a single "cluster network", even if it's spread across multiple CIDR blocks. So maybe this should be singular? ("`ClusterNetwork`") (And the struct type should not be called "ClusterNetwork" since it's only a *subset* of the cluster network.)

Reply:
That's a very good point; I was just duplicating what was done by openshift-sdn (in the interest of not rocking the boat too much). But you're right, this is a good chance to make this cleaner.

K8s is not the only one guilty of conflating the word "network" with "contiguous block of addresses."

Also, I know upstream calls it the ClusterCIDR, which makes me wonder if we should standardize on that. I don't like using the suffix "CIDR" to indicate an address block either.

Enough rambling. How about `ClusterNetwork []ClusterNetworkBlock`